core/trino-main/src/main/java/io/trino/operator/output/UnnestingPositionsAppender.java

```diff
@@ -63,6 +63,20 @@ public void appendRle(Block block, int rlePositionCount)
         delegate.appendRle(block, rlePositionCount);
     }
 
+    @Override
+    public void append(int position, Block source)
+    {
+        if (source instanceof RunLengthEncodedBlock runLengthEncodedBlock) {
+            delegate.append(0, runLengthEncodedBlock.getValue());
```

Comment:
Why not "appendRle" here?

Reply:
`appendRle` is less efficient in most cases in terms of CPU.
I did consider it because it has the upside of producing RLE block if all input blocks are RLE with the same value so network traffic would decrease but decided that overhead for every row is not worth it.
I did not run benchmarks for this though, I will do it because if the overhead is not that big producing RLE has its value.

Reply:
I ran JMH to compare `append` vs `appendRle` and the former is significantly slower (up to 3X)

```
| channelCount | enableCompression | nullRate | partitionCount | positionCount | type                  | append  | appendRle | appendRle % |
| ------------ | ----------------- | -------- | -------------- | ------------- | --------------------- | ------- | --------- | ----------- |
| 1            | FALSE             | 0        | 1000           | 10            | RLE_BIGINT            | 118.514 | 133.23    | 12.4170984  |
| 1            | FALSE             | 0        | 1000           | 10            | ROW_RLE_BIGINT_BIGINT | 77.396  | 81.651    | 5.49770014  |
| 1            | FALSE             | 0        | 1000           | 100           | RLE_BIGINT            | 152.432 | 226.563   | 48.63217697 |
| 1            | FALSE             | 0        | 1000           | 100           | ROW_RLE_BIGINT_BIGINT | 112.26  | 129.523   | 15.37769464 |
| 1            | FALSE             | 0        | 1000           | 500           | RLE_BIGINT            | 223.786 | 563.822   | 151.9469493 |
| 1            | FALSE             | 0        | 1000           | 500           | ROW_RLE_BIGINT_BIGINT | 210.256 | 363.192   | 72.73799559 |
| 1            | FALSE             | 0        | 1000           | 1000          | RLE_BIGINT            | 312.515 | 995.08    | 218.4103163 |
| 1            | FALSE             | 0        | 1000           | 1000          | ROW_RLE_BIGINT_BIGINT | 351.513 | 684.464   | 94.71939871 |
| 1            | FALSE             | 0        | 1000           | 1999          | RLE_BIGINT            | 492.314 | 1876.512  | 281.1616164 |
| 1            | FALSE             | 0        | 1000           | 1999          | ROW_RLE_BIGINT_BIGINT | 683.095 | 1299.389  | 90.22083312 |
| 1            | FALSE             | 0.2      | 1000           | 10            | RLE_BIGINT            | 118.375 | 145.382   | 22.81478353 |
| 1            | FALSE             | 0.2      | 1000           | 10            | ROW_RLE_BIGINT_BIGINT | 88.252  | 92.292    | 4.577799937 |
| 1            | FALSE             | 0.2      | 1000           | 100           | RLE_BIGINT            | 152.097 | 223.736   | 47.10086326 |
| 1            | FALSE             | 0.2      | 1000           | 100           | ROW_RLE_BIGINT_BIGINT | 98.02   | 119.2     | 21.60783514 |
| 1            | FALSE             | 0.2      | 1000           | 500           | RLE_BIGINT            | 223.654 | 566.095   | 153.1119497 |
| 1            | FALSE             | 0.2      | 1000           | 500           | ROW_RLE_BIGINT_BIGINT | 184.665 | 305.838   | 65.61774023 |
| 1            | FALSE             | 0.2      | 1000           | 1000          | RLE_BIGINT            | 313.747 | 1021.202  | 225.4858214 |
| 1            | FALSE             | 0.2      | 1000           | 1000          | ROW_RLE_BIGINT_BIGINT | 293.456 | 538.015   | 83.33753612 |
| 1            | FALSE             | 0.2      | 1000           | 1999          | RLE_BIGINT            | 490.783 | 1916.683  | 290.5357358 |
| 1            | FALSE             | 0.2      | 1000           | 1999          | ROW_RLE_BIGINT_BIGINT | 548.206 | 1061.718  | 93.67135712 |
```